src/models/cooling_effect.js

```diff
@@ -0,0 +1,169 @@
+ * @docname Cooling Effect (CE)
+ *
+ * @param {number} tdb - dry bulb air temperature, default in [°C] in [°F] if `units` = 'IP'
+ * @param {number} tr - mean radiant temperature, default in [°C] in [°F] if `units` = 'IP'
+ * @param {number} vr - relative air speed, default in [m/s] in [fps] if `units` = 'IP'
+ *
+ * Note: vr is the relative air speed caused by body movement and not the air
+ * speed measured by the air speed sensor. The relative air speed is the sum of the
+ * average air speed measured by the sensor plus the activity-generated air speed
+ * (Vag). Where Vag is the activity-generated air speed caused by motion of individual body parts.
+ * vr can be calculated using the function `v_relative` which is in .utilities.js.
+ *
+ * @param {number} rh - relative humidity, [%]
+ * @param {number} met - metabolic rate, [met]
+ * @param {number} clo - clothing insulation, [clo]
+ *
+ * Note: The activity as well as the air speed modify the insulation characteristics
+ * of the clothing and the adjacent air layer. Consequently, the ISO 7730 states that
+ * the clothing insulation shall be corrected {@link #ref_2|[2]}. The ASHRAE 55 Standard corrects
+ * for the effect of the body movement for met equal or higher than 1.2 met using
+ * the equation clo = Icl × (0.6 + 0.4/met) The dynamic clothing insulation, clo,
+ * can be calculated using the function `clo_dynamic` which is in .utilities.js.
+ *
+ * @param {number} [wme=0] - external work
+ * @param {'SI','IP'} [units= "SI"] - select the SI (International System of Units) or the IP (Imperial Units) system.
+ * @returns {number} ce - Cooling Effect, default in [°C] in [°F] if `units` = 'IP'
+ *
+ * @example
+ * const CE = cooling_effect(25, 25, 0.3, 50, 1.2, 0.5);
+ * console.log(CE); // Output: 1.64
+ *
+ * // For users who want to use the IP system
+ * const CE_IP = cooling_effect(77, 77, 1.64, 50, 1, 0.6, "IP");
+ * console.log(CE_IP); // Output: 3.74
+ */
+export function cooling_effect(
+  tdb,
+  tr,
+  vr,
+  rh,
+  met,
+  clo,
+  wme = 0,
+  units = "SI",
+) {
+  if (units.toLowerCase() === "ip") {
+    const result = units_converter({ tdb, tr, vr }, "IP");
+    tdb = result.tdb;
+    tr = result.tr;
+    vr = result.vr;
+  }
+
+  if (vr <= 0.1) {
+    return 0;
+  }
+
+  const still_air_threshold = 0.1;
+
+  const initial_set_tmp = set_tmp(
+    tdb,
+    tr,
+    vr,
+    rh,
+    met,
+    clo,
+    wme,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    false,
+    {
+      round: false,
+      calculate_ce: true,
+    },
+  );
+
+  function func(x) {
+    return (
+      set_tmp(
+        tdb - x,
+        tr - x,
+        still_air_threshold,
+        rh,
+        met,
+        clo,
+        wme,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        false,
+        {
+          round: false,
+          calculate_ce: true,
+        },
+      ) - initial_set_tmp
+    );
+  }
+
+  function findRoot(func, a, b) {
```

Comment:
declare this function outside this function and take a callback for `func`

Reply:
done